reconcile/utils/terrascript_client.py

```diff
@@ -4460,3 +4475,204 @@ def populate_tf_resource_route53_zone(self, resource, namespace_info):
+                    ],
+                    "Principal": {
+                        "Service": "cognito-idp.amazonaws.com",
+                    },
+                    "Condition": {
+                        "StringEquals": {
+                            "sts:ExternalId": self.get_values("sms_role_ext_id")
+                        }
+                    }
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "sns:Publish",
+                    ],
+                    "Resource": "*",
+                }
+            ]
+        }
+
+        sms_iam_role_resource = aws_iam_role(
+            "sms_role",
+            name=f'{identifier}-SMS',
+            description="role for applicant cognito, send sms",
+            assume_role_policy=sms_role_policy,
+            force_detach_policies=False,
+            max_session_duration=3600,
+            path="/service-role/",
+        )
+        tf_resources.append(sms_iam_role_resource)
+
+        # https://registry.terraform.io/providers/hashicorp/aws/3.60.0/docs/resources/cognito_identity_provider
+
+        pool_args = common_values.get("user_pool_properties", None)
+        pool_client_args = common_values.get("user_pool_client_properties", None)
+
+        if pool_args and pool_client_args:
+            pool_args_values = self.get_values('pool_args')
+            cognito_user_pool_resource = aws_cognito_user_pool(
+                "pool",
+                name=f'{identifier}-pool',
+                **pool_args_values
+            )
+            tf_resources.append(cognito_user_pool_resource)
+
+            cognito_user_pool_domain_resource = aws_cognito_user_pool_domain(
+                "userpool_domain",
+                domain=self.get_values("domain"),
+                user_pool_id=cognito_user_pool_resource.id
+            )
+            tf_resources.append(cognito_user_pool_domain_resource)
+
+            pool_client_args_values = self.get_values("user_pool_client_properties")
+            cognito_user_pool_client = aws_cognito_user_pool_client(
+                "userpool_client",
+                user_pool_id=cognito_user_pool_resource.id,
+                **pool_client_args_values
+            )
+            tf_resources.append(cognito_user_pool_client)
+        self.add_resources(account, tf_resources)
+
+    def populate_tf_resource_api_gateway(self, resource, namespace_info):
+        account, identifier, common_values, output_prefix, \
+            output_resource_name, annotations = \
+            self.init_values(resource, namespace_info)
+
+        tf_resources = []
+
+        self.init_common_outputs(tf_resources, namespace_info, output_prefix,
+                                 output_resource_name, annotations)
+
+        # FIXME
+        # I need to find an example of this pattern. Creating a dependent resource in an outside
+        # provider and referencing it here
```

Comment:
Grasping at straws a little here but potentially could reference the user_pool similar to [this example](https://github.com/app-sre/qontract-reconcile/blob/7b062e0983b421916cf736d258ecc5e35c86fc15/reconcile/utils/terrascript_client.py#L761). String literal `aws_cognito_user_pool` and concatenate the identifier?